core/src/main/java/hudson/security/PermissionScope.java

```diff
@@ -56,7 +58,7 @@
     /**
      * Domain model type that approximates this scope.
      */
-    public final Class modelClass;
+    public final Class<? extends ModelObject> modelClass;
```

Comment:
It is binary compatible, just not source compatible (without `unchecked` warnings anyway)